src/Stack/Storage.hs

```diff
@@ -151,6 +153,10 @@ CompilerCache
   globalDump Text
 
   UniqueCompilerInfo ghcPath
+
+-- History of checks for whether we should upgrade Stack
+UpgradeCheck
```

Comment:
If I understand the code correctly, this is being used to determine whether there have been any version checks in the previous 24 hours (before the current time).  It looks like this could grow indefinitely, and since there's no index on the `timestamp` field each of these checks would require a table scan, so over time this would get slower and slower.

I think you could add a unique constraint on `timestamp` (and change `logUpgradeCheck` to use `upsert`) to avoid the table scans, and I think records older than 24 hours could be purged since they're not being used anymore.

Alternatively, perhaps the approach could be simplified into doing just one version check on a given calendar date (rather than the last 24 hour period).  Then all you need to do is store a single value: the date of the last check.

Reply:
I'd like to skip the calendar day approach, since (1) it allows for users getting bugged twice in quick succession around midnight, and (2) it can be wonky with timezones. I'll update the PR to delete all of the old records before inserting. I left the old records initially because I thought it could be nice to see when checks actually happen, but there's 0 obvious reason to do it, just a bad data hording tendency on my part.